kong/response.go

```diff
@@ -37,3 +37,12 @@ func hasError(res *http.Response) error {
 	body, _ := io.ReadAll(res.Body) // TODO error in error?
 	return NewAPIError(res.StatusCode, messageFromBody(body))
 }
+
+func hasErrorRaw(res *http.Response) error {
+	if res.StatusCode >= 200 && res.StatusCode <= 399 {
+		return nil
+	}
+
+	body, _ := io.ReadAll(res.Body) // TODO error in error?
```

Comment:
Can we just wrap the error from this and put into the return value propagating whatever was returned by `io.ReadAll`? Otherwise this will get lost. Rarely but still, it can happen.